volttron/platform/vip/agent/subsystems/pubsub.py

```diff
@@ -101,154 +104,41 @@ def __init__(self, core, rpc_subsys, peerlist_subsys, owner):
         self._peer_subscriptions = {}
         self._my_subscriptions = {}
         self.protected_topics = ProtectedPubSubTopics()
+        core.register('pubsub', self._handle_subsystem, self._handle_error)
+        self.vip_socket = None
+        self._results = ResultsDictionary()
+        self._event_queue = Queue()
+        self._retry_period = 300.0
+        self._processgreenlet = None
 
         def setup(sender, **kwargs):
             # pylint: disable=unused-argument
-            rpc_subsys.export(self._peer_sync, 'pubsub.sync')
-            rpc_subsys.export(self._peer_subscribe, 'pubsub.subscribe')
-            rpc_subsys.export(self._peer_unsubscribe, 'pubsub.unsubscribe')
-            rpc_subsys.export(self._peer_list, 'pubsub.list')
-            rpc_subsys.export(self._peer_publish, 'pubsub.publish')
-            rpc_subsys.export(self._peer_push, 'pubsub.push')
+            self._processgreenlet = gevent.spawn(self._process_loop)
             core.onconnected.connect(self._connected)
-            core.onviperror.connect(self._viperror)
             peerlist_subsys.onadd.connect(self._peer_add)
-            peerlist_subsys.ondrop.connect(self._peer_drop)
-
+            self.vip_socket = self.core().socket
             def subscribe(member):   # pylint: disable=redefined-outer-name
                 for peer, bus, prefix in annotations(
                         member, set, 'pubsub.subscriptions'):
                     # XXX: needs updated in light of onconnected signal
                     self.add_subscription(peer, prefix, member, bus)
+                    #_log.debug("PUBSUB SUBSYS subscribe with annotate. prefix {0}, identity: {1}".format(prefix, self.core().identity))
```

Comment:
Either uncomment debug messages or remove them.  I would be in favor of uncommented even if it's more traffic in the logs.